library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -702,6 +697,28 @@ static String getWhereClauseForIds(long[] ids) {
     }
 
     /**
+     * Enqueue a new download batch.
+     *
+     * @param batch the parameters specifying this batch
+     * @return an ID for the batch, unique across the system.  This ID is used to make future
+     * calls related to this batch.
+     */
+    public long enqueue(RequestBatch batch) {
+        long batchId = insert(batch);
+        for (Request request : batch.getRequests()) {
+            request.setBatchId(batchId);
```

Comment:
Even if we set the `requestid` for each request inside the batch, we will anyway have to do a loop here in order to `insert` each request into the db unless we want to do that on the batch as well, but that seems a little bit too much responsibility for the `RequestBatch`.